src/components/Root/token-util.js

```diff
@@ -212,6 +214,27 @@ export const isRotating = () => {
 export const rtr = (fetchfx, logger, callback, okapi) => {
   logger.log('rtr', '** RTR ...');
 
+  /**
+   * Check the windowId against the active windowId stored in sessionStorage.
+   * If they don't match, this window is not the active one, so we skip the
+   * rotation request and return a promise that resolves
+   * with the current token expiry data. This is to prevent multiple windows
+   * from trying to rotate the token at the same time.
+   */
+  const thisWindowId = window.stripesRTRWindowId;
```

Comment:
I'm confused about this logic - aren't `window` and `sessionStorage` tab-specific, and so comparing them would always be equal?

Reply:
`window.stripesRTRWindowId` stores the UUID of each window - it's only set once per tab and does not change.  
Session storage is updated across tabs when a new tab is focused via `BroadcastChannel` sending out the handled `message` to the other tabs to update their `SessionStorage`value with the `stripesRTRWindowId` of the active window.